rotkehlchen/chain/evm/decoding/extrafi/cache.py

```diff
@@ -0,0 +1,115 @@
+from collections.abc import Iterator, Sequence
+from itertools import islice
+from typing import TYPE_CHECKING, Literal
+
+from eth_utils import to_checksum_address
+
+from rotkehlchen.chain.evm.contracts import EvmContract
+from rotkehlchen.chain.evm.decoding.extrafi.constants import EXTRAFI_POOL_CONTRACT
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.globaldb.cache import (
+    globaldb_get_general_cache_values,
+    globaldb_get_unique_cache_value,
+    globaldb_set_general_cache_values_at_ts,
+    globaldb_set_unique_cache_value_at_ts,
+)
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.types import CacheType, ChainID, ChecksumEvmAddress
+from rotkehlchen.utils.misc import ts_now
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+
+def chunk_range(iterable: Sequence, size: int) -> Iterator:
+    """Split into chunks of size `size` the iterable provided"""
+    it = iter(iterable)
+    return iter(lambda: tuple(islice(it, size)), ())
+
+
+def get_existing_reward_pools(chain_id: ChainID) -> tuple[set[ChecksumEvmAddress]]:
+    """Get extrafi rewards pools for extrafi lending"""
+    with GlobalDBHandler().conn.read_ctx() as cursor:
```

Comment:
pass the cursor to this function instead. At one place where you use it you initialize a cursor right before calling. That's the cases where you know you should pass the cursor as an argument.

Reply:
It is manually called only in the tests. In the actual code it is called at `reload_data` that doesn't reuse the cursor